### PR TITLE
Add new "metrics" API group.

### PR DESCRIPTION
**Pull request by timstclair**
Forked from https://github.com/kubernetes/kubernetes/pull/16458 for the purposes of separating the discussion of the API group setup from the discussion of the API itself. I added 2 "placeholder" types (which will eventually be fleshed out) for purposes of demonstrating the API setup, and giving the code-generation tools something to generate.

From the [metrics group proposal](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/compute-resource-metrics-api.md)
First steps towards https://github.com/kubernetes/kubernetes/issues/12483

cc/ @vishh @dchen1107 @lavalamp

### CODE REVIEW

File: pkg/api/testapi/testapi.go [hunk 1]
```diff
@@ -90,21 +91,11 @@ func (g TestGroup) GroupAndVersion() string {
...
-		}
-		return interfaces.Codec
-	}
-	if g.Group == "extensions" {
-		interfaces, err := latest.GroupOrDie("extensions").InterfacesFor(g.GroupVersionUnderTest)
-		if err != nil {
-			panic(err)
-		}
-		return interfaces.Codec
+	interfaces, err := latest.GroupOrDie(g.Group).InterfacesFor(g.GroupVersionUnderTest)
```
Comment (caesarxuchao) [inline pkg/api/testapi/testapi.go]: Thank you for fixing this!

**Comment by vishh** [comment]
LGTM.

**Comment by caesarxuchao** [comment]
Probably you need to import it the install package in https://github.com/kubernetes/kubernetes/blob/master/pkg/master/import_known_versions.go as well. LGTM otherwise.

**Comment by timstclair** [comment]
I think the first iteration (just Raw{Node,Pod,Container}) will only be served by the kubelet, so I don't think we want to import it to the master? What does importing to the master do?

**Comment by caesarxuchao** [comment]
Per offline-discussion, master doesn't need to know the metric API, so no need to import it there. Thanks. LGTM.

**Comment by wojtek-t** [comment]
@timstclair - can you please rebase?

**Comment by timstclair** [comment]
Rebased.

**Comment by timstclair** [comment]
@k8s-bot unit test this

File: pkg/api/testapi/testapi.go (hunk 1 above)
Comment (lavalamp) [inline pkg/api/testapi/testapi.go]: +1

**Comment by lavalamp** [comment]
LGTM

**Pull request merged**